nssdb/keys.go

```diff
@@ -227,3 +240,9 @@ func (db *NSSDB) AddPublicKey(ctx context.Context, pubKey *ecdsa.PublicKey, ckaI
 
 	return pubKeyID, nil
 }
+
+var printableRx = regexp.MustCompile(`^[a-zA-Z0-9 '()+,\-.\\:=?]*$`)
+
+func isPrintable(s string) bool {
+	return printableRx.MatchString(s)
+}
```

Comment:
Is this a limitation of nssdb? 

The [unicode](https://pkg.go.dev/unicode) package has a method for validating whether a run is printable [IsPrint(rune)](https://pkg.go.dev/unicode#IsPrint). Some more complicated limitations can be done using the [In(r rune, ranges ...*RangeTable)](https://pkg.go.dev/unicode#In) function, for example, to limit the runes to be in the `Latin` [RangeTable](https://pkg.go.dev/unicode#RangeTable), or you can also create your own.

Reply:
This refers to the asn.1 type of printable strings, which is more restrictive than the unicode IsPrint function. In a certificate where the common name contains only printable characters it's encoded with the asn1 tag for PrintableString. For anything else it gets the asn1 tag for UTF8String. (At least that's how it works for our certificates.)

Reply:
Looping runes will perform better:
https://github.com/smallstep/crypto/blob/9fd47c3cda1afa9da212dd49dbe24622d1482b85/x509util/utils.go#L183-L213

Go ans1 loops around bytes using a similar method, this is for one byte https://github.com/golang/go/blob/f24b299df2896a4e8a80863dbb55a264f4b9bb68/src/encoding/asn1/asn1.go#L422-L444

Reply:
9ce1394